pkg/collector/python/test_loader.go

```diff
@@ -206,3 +224,47 @@ func testLoadWheelCheck(t *testing.T) {
 	// test we call get_attr_string on the module
 	assert.Equal(t, C.get_attr_string_py_class, C.get_class_dd_wheel_py_module)
 }
+
+func testLoadHACheck(t *testing.T) {
+	C.reset_loader_mock()
+
+	conf := integration.Config{
+		Name:       "fake_check",
+		Instances: []integration.Data{integration.Data("{\"value\": 1}")},
+		InitConfig: integration.Data("{}"),
+	}
+
+	rtloader = newMockRtLoaderPtr()
+	defer func() { rtloader = nil }()
+	senderManager := mocksender.CreateDefaultDemultiplexer()
+	logReceiver := option.None[integrations.Component]()
+	tagger := nooptagger.NewComponent()
+	loader, err := NewPythonCheckLoader(senderManager, logReceiver, tagger)
+	assert.Nil(t, err)
+
+	pkgconfigsetup.Datadog().SetWithoutSource("ha_agent.enabled", true)
+
+	// testing loading custom checks
+	C.get_class_return = 1
+	C.get_class_py_module = newMockPyObjectPtr()
+	C.get_class_py_class = newMockPyObjectPtr()
+	C.get_attr_string_return = 0
+	C.get_check_return = 0
+	C.get_check_deprecated_check = newMockPyObjectPtr()
+	C.get_check_deprecated_return = 1
+
+	C.get_attr_bool_return = 1
+	C.get_attr_bool_attr_value = 1
+
+	check, err := loader.Load(senderManager, conf, conf.Instances[0])
+	// Remove check finalizer that may trigger race condition while testing
+	runtime.SetFinalizer(check, nil)
+
+	assert.Nil(t, err)
+	assert.Equal(t, "fake_check", check.(*PythonCheck).ModuleName)
+	assert.Equal(t, "unversioned", check.(*PythonCheck).version)
+	assert.Equal(t, C.get_class_py_class, check.(*PythonCheck).class)
+	// test we call get_attr_bool on the class
+	assert.Equal(t, C.get_attr_bool_py_class, C.get_class_py_class)
+	assert.Equal(t, true, check.(*PythonCheck).haSupported)
```

Comment:
Can we also test the other code path / cases?
- when `ha_agent.enabled: false`
- when `C.get_attr_bool` returns an error
- when `C.get_attr_bool` return false value for C.get_attr_bool_attr_value
- when `C.get_attr_bool` return true value for C.get_attr_bool_attr_value

Using a go subtest / table tests